### ISSUE TITLE
Startup issues with the dtrt-indent package

### ISSUE THREAD

**Issue by 3rd3** [syl20bnr/spacemacs#9437]
#### Description :octocat:
Installing and enabling the `dtrt-indent` package results in a startup error.

#### Reproduction guide :beetle:
- Start Emacs
- Install `dtrt-indent` via `dotspacemacs-additional-packages '(dtrt-indent)`.
- Enable it via  `M-x customize-variable dtrt-indent-mode`, press ``TOGGLE`` and `Save and apply`.
- Open a C++ file and observe that the module correctly infers the indentation settings of the given file.
- Restart Emacs

*Observed behaviour:* :eyes: :broken_heart:

    Error loading .spacemacs: (file-error "Cannot open load file" "No such file or directory" "dtrt-indent")

*Expected behaviour:* :heart: :smile:

Emacs should start normally.

#### System Info :computer:
- OS: darwin
- Emacs: 25.2.1
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af14)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp syntax-checking auto-completion semantic
            (latex :variables latex-enable-auto-fill t)
            org git github version-control extra-langs markdown emacs-lisp haskell html lua php javascript shell-scripts java
            (haskell :variables haskell-enable-ghc-mod-support nil haskell-enable-ghci-ng-support t haskell-enable-hindent-style "gibiansky" haskell-enable-shm-support nil)
            (c-c   :variables c-c  -enable-clang-support t)
            (python :variables python-test-runner 'pytest python-fill-column 99)
            (clojure :variables clojure-enable-fancify-symbols t)
            (shell :variables shell-default-height 20 shell-default-position 'bottom shell-enable-smart-eshell t shell-default-shell 'eshell)
            evil-snipe)
```
- System configuration features: JPEG NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS

**Comment by 3rd3** [comment]
The toggle adds the following setting to `(custom-set-variables ...)` in .spacemacs`:

     '(dtrt-indent-mode t nil (dtrt-indent))

**Comment by 3rd3** [comment]
Maybe @jscheid has an idea?

**Comment by 3rd3** [comment]
I am using this fix for now:

```
  (mapc (lambda (x) (add-to-list 'load-path x))
        (file-expand-wildcards "~/.emacs.d/elpa/dtrt-indent*"))
  (require 'dtrt-indent)
  (dtrt-indent-mode 1)
```

**Comment by jscheid** [comment]
I don't maintain dtrt-indent anymore and haven't had anything to do with packaging for melpa, so can't be of help. If you think it's an issue with the upstream package you might want to file a ticket over at https://github.com/jscheid/dtrt-indent, but I think it more polite to wait and see what spacemacs people have say before escalating further. I'm not aware of other people having troubles with the melpa package.

**Comment by brandon-rhodes** [comment]
For anyone coming to this issue from a Google search: I encountered this error because my `custom-set-variables` stanza is at the top of my `init.el`, above things like `(add-to-list 'load-path "~/.emacs.d/site-lisp")` that would make it possible to import packages yet. And it appears that dtrt-indent’s preferred way of getting itself loaded is via a customize setting `dtrt-indent-mode` that has the side effect of trying to load the mode itself.

I therefore skipped the customize variable it wants to use, and instead put this at the bottom of my `init.el`:

    (dtrt-indent-global-mode)

As long as this is after I've set up my `load-path`, it seems to work fine.
